hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirConcatOp.java

```diff
@@ -184,6 +185,12 @@ private static void verifyTargetFile(FSDirectory fsd, final String target,
         throw new HadoopIllegalArgumentException("concat: the src file " + src
             + " is the same with the target file " + targetIIP.getPath());
       }
+
+      if(srcINodeFile.getStoragePolicyID() == HdfsConstants.DB_STORAGE_POLICY_ID) {
```

Comment:
I thought that if we tried to concat to a small file the file was moved to normal storage and then concatenated to, is it something that should have already happened when we reach this point or is it completly forbiden to call concat on a small file?

Reply:
Can not concat small files. I have put the check here as all other checks are done in this fn

Reply:
So is this a new behavior? Or what did we do before?

Reply:
Previously, it did not work. This change will throw a proper exception

Reply:
I'll open a jira to support concat for small files.